R/thicknessOf.R

```diff
@@ -29,24 +30,26 @@
 #' thicknessOf(jacobs2000, "Bt")
 #' 
 #' # maximum bottom depth minus minimum top depth of horizon designations matching "Bt"
-#' thicknessOf(jacobs2000, "Bt", method = "minmax")
+#' thicknessOf(jacobs2000, "Bt", prefix = "Bt_", method = "minmax")
 #' 
 #' # cumulative thickness of horizon designations matching "A|B"
-#' thicknessOf(jacobs2000, "A|B")
+#' thicknessOf(jacobs2000, "A|B", prefix = "AorB_")
 #' 
 #' # maximum bottom depth minus minimum top depth of horizon designations matching "A|B"
-#' thicknessOf(jacobs2000, "A|B", method = "minmax")
+#' thicknessOf(jacobs2000, "A|B", method = "minmax", prefix = "AorB_")
 #' # note that the latter includes the thickness of E horizons between the A and the B
 #' 
 #' # when using a custom function (be sure to accept ... and consider the effect of NA values)
 #' 
 #' # calculate cumulative thickness of horizons containing >18% clay
-#' thicknessOf(jacobs2000, FUN = function(x, ...) !is.na(x[["clay"]]) & x[["clay"]] > 18)
+#' thicknessOf(jacobs2000, prefix = "claygt18_", 
+#'             FUN = function(x, ...) !is.na(x[["clay"]]) & x[["clay"]] > 18)
 #' 
 thicknessOf <- function(x, 
                         pattern = NULL, 
                         hzdesgn = guessHzDesgnName(x, required = TRUE), 
```

Comment:
I can't remember if we wanted to default to guessing (across all functions) when horizon names aren't specified or set in metadata. Currently there is a mixture of approaches across aqp.

Reply:
I know you have more recently decided to **not** guess (which has broken or changed output in several places), but I don't think there is a rule we agreed on. If you are making one then we can start a change here.

The functions for guessing horizon designation, texture class names, and generic attributes are used in several places (pretty much all "my" functions I suppose) I could replace all uses of guessxx() with the metadata getter, and required=TRUE argument, but for now this is consistent with many of my other function usages. Let's address that (deprecating the 3 guess functions) in a separate issue or PR.

Reply:
I have started to address this in #309, and will make a corresponding fix in this PR. Thanks for prodding me on this, there was lots of room to improve consistency and I am happy to tackle it across all functions.